### ISSUE TITLE
istioctl manifest generates objects of removed apis

### ISSUE THREAD

**Issue by infa-ddeore** [istio/istio#43117]
### Bug Description

istioctl manifest generate command generates `policy/v1beta1` object which is removed in 1.25 k8 so `PodDisruptionBudget` object cant be applied in 1.25 k8

```
istio-1.16.2/bin/istioctl manifest generate --set profile=minimal | grep -B1 "kind: PodDisruptionBudget"
apiVersion: policy/v1beta1
kind: PodDisruptionBudget
```

same for `HorizontalPodAutoscaler` object which is removed in `1.26`
```
istio-1.16.2/bin/istioctl manifest generate --set profile=minimal | grep -B1 HorizontalPodAutoscaler
apiVersion: autoscaling/v2beta2
kind: HorizontalPodAutoscaler
```

### Version

```prose
NA
```

### Additional Information

this affects istio control plane installation, workaround is to use `istioctl install --set profile=minimal` 

but it also affects user gateway deployment via `istioctl manifest generate -f user-gw-override.yaml` command

**Comment by howardjohn** [comment]
there is auto detection of support if you set a flag (I think it's --detect-cluster or similar). We may need to change the default as well though

**Comment by infa-ddeore** [comment]
thanks for the quick reply @howardjohn 

its `--cluster-specific` flag available in 1.16

```
istio-1.16.2/bin/istioctl manifest generate --set profile=minimal --cluster-specific |grep -B1 -E 'kind: PodDisruptionBudge|HorizontalPodAutoscaler' |grep api
apiVersion: policy/v1
apiVersion: autoscaling/v2
```

do you know if same can be achieved with 1.14 istioctl so the api deprecation warnings will be removed when using 1.14 istioctl binary on 1.24 or lower versions?

**Comment by hanxiaop** [comment]
> 

@infa-ddeore 1.14 cannot do it, and it is EOL. Why not use 1.16 istioctl? If you want to use 1.14 manifests, you can specify the manifests path using the `--manifests` flag.

**Comment by infa-ddeore** [comment]
1.16 is adding `credential-socket` as extra mount, rest remains same as 1.14 except the new apis, not sure if this extra mount will create problem or not. Also 1.16 `--cluster-specific` option needs access to k8 cluster, without that it doesn't run

```
$ diff <(~/tmp/istio-1.14.3/bin/istioctl manifest generate -f user-gw-override.yaml) <(~/tmp/istio-1.16.2/bin/istioctl manifest generate --cluster-specific -f user-gw-override.yaml)
121c121
<         image: docker.io/istio/proxyv2:1.14.3
---
>         image: docker.io/istio/proxyv2:1.16.2
166a167,168
>         - mountPath: /var/run/secrets/credential-uds
>           name: credential-socket
200a203,204
>         name: credential-socket
>       - emptyDir: {}
236c240
< apiVersion: policy/v1beta1
---
> apiVersion: policy/v1
290c294
< apiVersion: autoscaling/v2beta2
---
> apiVersion: autoscaling/v2
```

as a workaround i will use 1.14 istioctl and `sed` find/ replace `apiVersion` on generated manifest

@howardjohn making `--cluster-specific` as default will create problem because it needs istioctl to connect to k8 cluster, instead can an extra argument be introduces to provide k8 version to tell istioctl which k8 version the manifests are intended to?

eg.
```
istioctl manifest generate --set profile=minimal --cluster-specific --k8-version 1.25
```

**Comment by infa-ddeore** [comment]
> I don't mean making cluster-specific default, that will always be optional. I mean the default version we assume k8s is if they do not use that

that makes sense, default version to assume need to be higher to generate objects with newer apis

@howardjohn should i create another issue to increase default k8 version that istioctl assumes or its already in progress?

**Comment by hanxiaop** [comment]
I think once the min k8s version is updated correctly, we won't have this issue. `manifest generate` uses minimum supported k8s version as the default to generate charts. For example, there won't be any `policy/v1beta1` charts after https://github.com/istio/istio/pull/41838 get merged.

**Comment by infa-ddeore** [comment]
thanks @hanxiaop

**Comment by infa-ddeore** [comment]
> I think once the min k8s version is updated correctly, we won't have this issue. `manifest generate` uses minimum supported k8s version as the default to generate charts. For example, there won't be any `policy/v1beta1` charts after #41838 get merged.

@hanxiaop #41838 is merged but it still generates deprecated objects with `1.15.5` and `1.16.2`, am I missing something here?

```
$ ./istio-1.15.5/bin/istioctl manifest generate --set profile=minimal | grep -B1 -E 'kind: PodDisruptionBudge|HorizontalPodAutoscaler' |grep apiVersion
apiVersion: policy/v1beta1
apiVersion: autoscaling/v2beta2
```

**Comment by hanxiaop** [comment]
@infa-ddeore It should be available in the next 1.16 release. FYI https://github.com/istio/istio/pull/43033
